pulsar-broker-common/src/main/java/org/apache/pulsar/bookie/rackawareness/BookieRackAffinityMapping.java

```diff
@@ -133,6 +137,28 @@ public synchronized void setConf(Configuration conf) {
         }
     }
 
+    private void watchAvailableBookies() {
+        BookieAddressResolver bookieAddressResolver = getBookieAddressResolver();
+        if (bookieAddressResolver instanceof DefaultBookieAddressResolver) {
+            try {
+                Field field = DefaultBookieAddressResolver.class.getDeclaredField("registrationClient");
+                field.setAccessible(true);
+                RegistrationClient registrationClient = (RegistrationClient) field.get(bookieAddressResolver);
+                registrationClient.watchWritableBookies(versioned -> {
+                    try {
+                        racksWithHost = bookieMappingCache.get(BOOKIE_INFO_ROOT_PATH).get()
```

Comment:
just wondering if this will block the meta thread without timeout?

Reply:
It's pure memory operation, won't block

Reply:
No, It will load data from metadata store while cache missed.

Reply:
It's in 
```
        this.executor = Executors
                .newSingleThreadScheduledExecutor(new DefaultThreadFactory("pulsar-registration-client"));
```
Won't block the zk thread.

Reply:
Any reason that prefer to use `.get()` to make a sync call?
As I understand, it just process the notifications from the `RegistrationClient`.
Looks like use async API is better?

Reply:
PulsarRegistrationClient holds another watcher. 
org.apache.bookkeeper.client.BookieWatcherImpl#initialBlockingBookieRead
```
                writable = this.registrationClient.watchWritableBookies((bookies) -> {
                    this.processWritableBookiesChanged((Set)bookies.getValue());
                });
```
We need to make the watcher callback ordered. First to update the rack info, then to handle `processWritableBookiesChanged`. Or the newBookieNode didn't aware of the newest rack info.
So I change the `writableBookiesWatchers ` type from `Map` to `Set`.

Reply:
Ah, got it. I though you want to remove the unused `Boolean.TRUE` 😂